frontend/src/features/mapSideBar/SidebarContents/AssociatedLandForm.tsx

```diff
@@ -532,12 +532,16 @@ const AssociatedLandForm: React.FC<IAssociatedLandParentForm> = (
           onSubmit={async (values, actions) => {
             const apiValues = valuesToApiFormat(_.cloneDeep(values), keycloak.agencyId);
             try {
+              let land;
               if (!values.data.id) {
-                await createBuilding(apiValues)(dispatch);
+                land = await createBuilding(apiValues)(dispatch);
               } else {
-                await updateBuilding(apiValues)(dispatch);
+                land = await updateBuilding(apiValues)(dispatch);
+              }
+              actions.resetForm({ values: { ...values, data: land } });
```

Comment:
reset the form so that closing the form doesn't trigger the unsaved changes warning.